examples/mouse_to_tile.rs

```diff
@@ -0,0 +1,383 @@
+                TilemapType::Hexagon(HexCoordSystem::RowEven) => {
+                    *map_type = TilemapType::Hexagon(HexCoordSystem::RowOdd);
+                }
+                TilemapType::Hexagon(HexCoordSystem::RowOdd) => {
+                    *map_type = TilemapType::Hexagon(HexCoordSystem::Column);
+                    *map_texture = TilemapTexture((*tile_handle_hex_col).clone());
+                    *tile_size = TILE_SIZE_HEX_COL;
+                    *grid_size = GRID_SIZE_HEX_COL;
+                }
+                TilemapType::Hexagon(HexCoordSystem::Column) => {
+                    *map_type = TilemapType::Hexagon(HexCoordSystem::ColumnEven);
+                }
+                TilemapType::Hexagon(HexCoordSystem::ColumnEven) => {
+                    *map_type = TilemapType::Hexagon(HexCoordSystem::ColumnOdd);
+                }
+                TilemapType::Hexagon(HexCoordSystem::ColumnOdd) => {
+                    *map_type = TilemapType::Square {
+                        diagonal_neighbors: false,
+                    };
+                    *map_texture = TilemapTexture((*tile_handle_square).clone());
+                    *tile_size = TILE_SIZE_SQUARE;
+                    *grid_size = GRID_SIZE_SQUARE;
+                }
+            }
+
+            let grid_size_vec: Vec2 = (*grid_size).into();
+            let label_offset = get_label_offset(&map_type) * grid_size_vec;
+
+            for (tile_pos, mut tile_label_transform) in tile_label_q.iter_mut() {
+                let tile_center = tile_pos.center_in_world(&grid_size, &map_type);
+                let text_center = tile_center + label_offset;
+                *tile_label_transform = *map_transform;
+                tile_label_transform.translation += text_center.extend(2.0);
+            }
+
+            for window in windows.iter() {
+                for (mut label_text, mut label_transform) in map_type_label_q.iter_mut() {
+                    *label_transform = Transform {
+                        translation: Vec2::new(
+                            -0.5 * window.width() / 2.0,
+                            0.8 * window.height() / 2.0,
+                        )
+                        .extend(0.0),
+                        ..Default::default()
+                    };
+                    *label_text = Text::from_section(
+                        format!("{:?}", map_type.as_ref()),
+                        TextStyle {
+                            font: font_handle.clone(),
+                            font_size: 20.0,
+                            color: Color::BLACK,
+                        },
+                    )
+                    .with_alignment(TextAlignment::CENTER);
+                }
+            }
+        }
+    }
+}
+
+#[derive(Component)]
+struct HighlightedLabel;
+
+pub fn cursor_pos_as_world_pos(
+    window: &Window,
+    camera_q: &Query<(&Transform, &Camera)>,
+) -> Option<Vec2> {
+    window.cursor_position().map(|cursor_pos| {
```

Comment:
I think we should likely pass cursor position directly into this using bevy events. It's a bit more code to write out but I think it's the recommended way of dealing with mouse input in bevy. See:
https://github.com/bevyengine/bevy/blob/main/examples/input/mouse_input_events.rs